### PR TITLE
middleware fix and improvement

### PR DESCRIPTION
**Pull request by mightqxc**

### CODE REVIEW

File: pandaharvester/harvestercore/spec_base.py [hunk 1]
```diff
@@ -16,9 +18,11 @@
 # encoder for non-native json objects
 class PythonObjectEncoder(json.JSONEncoder):
     def default(self, obj):
-        if isinstance(obj, (set, dict, list, bytearray)):
-            return json.JSONEncoder.default(self, obj)
-        return {'_non_json_object': pickle.dumps(obj)}
+        if isinstance(obj, rpyc.core.netref.BaseNetref):
+            retVal = rpyc.utils.classic.obtain(obj)
+        else:
+            retVal = {'_non_json_object': pickle.dumps(obj_to_encode)}
```
Comment (tmaeno) [inline pandaharvester/harvestercore/spec_base.py]: Shouldn't obj_to_encode be just obj?

Comment (mightqxc) [reply pandaharvester/harvestercore/spec_base.py]: fixed

File: pandaharvester/harvestercore/plugin_factory.py [hunk 2]
```diff
@@ -49,6 +49,6 @@ def get_plugin(self, plugin_conf):
         impl = cls(**args)
         # bare instance when middleware is used
         if 'original_config' in plugin_conf:
-            bare_impl = self.get_plugin(plugin_conf['original_config'])
+            bare_impl = plugin_conf['original_config']
```
Comment (tmaeno) [inline pandaharvester/harvestercore/plugin_factory.py]: I think that the following line

https://github.com/PanDAWMS/panda-harvester/blob/master/pandaharvester/harvestermiddleware/rpc_herder.py#L27

needs to be changed accordingly if the dict is set to self.bare_impl.

Comment (mightqxc) [reply pandaharvester/harvestercore/plugin_factory.py]: solved

**Pull request merged**